embeddings/src/lib.rs

```diff
@@ -38,97 +34,189 @@ fn handle_embeddings(req: Request) -> Result<Response> {
-        .unwrap())
-}
 
-fn delete_embeddings(_req: Request, _params: Params) -> Result<Response> {
-    // Delete the record with id
+    for (e, res) in embedding_request.iter().zip(embedding_result.embeddings) {
+        let vec = json!(res.clone());
+        let blob = serde_json::to_vec(&vec)?;
 
-    Ok(http::Response::builder()
-        .status(http::StatusCode::OK)
-        .body(None)
-        .unwrap())
-}
+        let query_params = [
+            sqlite::ValueParam::Text(e.reference.as_deref().unwrap_or_default()),
+            sqlite::ValueParam::Text(e.text.as_str()),
+            sqlite::ValueParam::Blob(blob.as_slice()),
+        ];
 
-#[derive(Debug, Serialize, Deserialize)]
-struct Embedding {
-    id: u32,
-    reference: String,
-    text: String,
-    embedding: Option<Vec<f32>>,
+        let _ = conn.execute(
+            "INSERT INTO embeddings ('reference', 'text', 'embedding') VALUES (?, ?, ?);",
+            &query_params,
+        );
+    }
+
+    Ok(embedding_request.len())
 }
 
 impl<'a> TryFrom<sqlite::Row<'a>> for Embedding {
     type Error = anyhow::Error;
 
     fn try_from(row: sqlite::Row<'a>) -> std::result::Result<Self, Self::Error> {
-        info!("DB Row: {:?}", row.get::<&str>("embedding"));
-        info!("DB Row: {:?}", row.get::<u32>("embedding"));
-        let id = row.get::<u32>("id").unwrap();
+        let id = Some(row.get::<u32>("id").unwrap_or_default());
         let reference = row
             .get::<&str>("reference")
             .context("reference column is empty")?;
         let text = row.get::<&str>("text").context("text column is empty")?;
-        let embedding = row.get::<&str>("embedding").unwrap_or_default();
-        info!("Embedding from DB {:?}", embedding);
-        //let embedding: Option<Vec<f32>> = embedding.unwrap()
-        //let embedding: Vec<f32> = bincode::deserialize_from(Cursor::new(embedding))?;
+        let embedding: Vec<f32> = match row.get::<&ValueResult>("embedding").unwrap() {
+            ValueResult::Blob(b) => {
+                serde_json::from_value(serde_json::from_slice(b.as_slice()).unwrap_or_default())?
+            }
+            _ => todo!(),
+        };
         Ok(Self {
             id,
-            reference: reference.to_owned(),
+            reference: Some(reference.to_owned()),
             text: text.to_owned(),
-            embedding: None,
+            embedding: Some(embedding),
         })
     }
 }
+
+fn get_similar(query: Vec<&str>) -> Result<SimilarityResultSet> {
+    let sql_query = "SELECT * FROM embeddings";
+    let conn = Connection::open_default()?;
+    let embedding_records: Vec<Embedding> = conn
+        .execute(sql_query, &[])?
+        .rows()
+        .map(|row| -> anyhow::Result<Embedding> { row.try_into() })
+        .collect::<anyhow::Result<Vec<Embedding>>>()?;
+    info!("String to compare: {:?}", query[0].to_string());
+    let mut result_set = SimilarityResultSet {
+        text: query[0].to_string(),
```

Comment:
I would not instantiate this up here. Collect the results into a vector in the loop below and then instantiate the `SimilarityResultSet` at the end of the function